build/gen_check.sh

```diff
@@ -17,40 +17,22 @@
 #!/usr/bin/env bash
 set -e
 
-function test_gen() {
-  local cmd="${1}"
-  local reference_dir="${2}"
-  shift 2
+# Copy the current examples dir
+tmp="$(mktemp -d "/tmp/tmp-gen_check.XXXXXX")"
+trap "rm -rf '${tmp}'" EXIT INT TERM RETURN ERR
+cp -ar './examples/' "${tmp}"
 
-  # Generate in a tmp dir, then diff with examples.
-  # Give the tmpdir a more meaningful name
-  suffix="$(echo "${reference_dir}" | cut -d'/' -f2,4 --output-delimiter='-')"
-  tmp="$(mktemp -d "/tmp/tmp-${suffix}.XXXXXX")"
-  trap "rm -rf '${tmp}'" EXIT INT TERM RETURN ERR
-
-  ${cmd} "${tmp}"
+# Regenerate examples
+./scripts/regen.sh
```

Comment:
can we regen it to a tmp location instead of polute the source?

Reply:
Done. Added an optional argument to `regen.sh`.